<h2>Description</h2>
<p>
This track represents the gene models for the assembly generated using the CAT
(Compartive Annotation Toolkit) software.  The reference annotations are from
GENCODE V38 mapped with the Minigraph-Cactus T2T-CHM13-based genomic alignments.
</p>

<h2>Display Conventions and Configuration</h2>
<p>
  This track follows the display conventions for
  <a href="https://genome.ucsc.edu/goldenPath/help/hgTracksHelp.html#GeneDisplay">gene
    prediction tracks</a>. The exons for putative non-coding genes and
  untranslated regions are represented by relatively thin blocks, while those
  for coding open reading frames are thicker. Gene names are displayed in 'pack'
  or 'full' mode. More information about each gene can be found by clicking on
  the specific gene/transcript model.
</p>
<p>
The following color key is used:
</p>
<ul>
  <li><span bgcolor="rgb(255,51,255)">Blue</span>: protein coding</li>
  <li><span bgcolor="rgb(85,212,76)">Green</span>: non-coding</li>
  <!-- <li><span>Purple</span>: novel predictions from PacBio Iso-Seq</li>  -->
  <li><span bgcolor="rgb(76,85,212">Pink</span>: pseudogenes
</ul>

<h2>Methods</h2>
<p>
Genome annotation for was performed using Comparative Annotation Toolkit (CAT). CAT
leverages whole-genome alignments generated by Minigrahp-Cactus to transfer annotations
from one source genome to one or more target genomes. For this annotation set,
CAT lifted over the reference GENCODE V38 annotations this HPRC genome.
The mapped transcripts were refined with AUGUSTUS to correct for genomic variation
and assembly errors.
</p>
<h2>Credits</h2>
<p>
  This track was provide by Marina Haukness <a href="mailto:mhauknes@ucsc.edu">&lt;mhauknes@ucsc.edu&gt;</a>
  of UC Santa Cruz.
</p>

<h2>References</h2>
<p>
Fiddes IT, Armstrong J, Diekhans M, Nachtweide S, Kronenberg ZN, Underwood JG, Gordon D, Earl D,
Keane T, Eichler EE <em>et al</em>.
<a href="https://www.ncbi.nlm.nih.gov/pubmed/29884752" target="_blank">
Comparative Annotation Toolkit (CAT)-simultaneous clade and personal genome annotation</a>.
<em>Genome Res</em>. 2018 Jul;28(7):1029-1038.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29884752" target="_blank">29884752</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6028123/" target="_blank">PMC6028123</a>
</p>
<p>
Stanke M, Diekhans M, Baertsch R, Haussler D.
<a href="https://www.ncbi.nlm.nih.gov/pubmed/18218656" target="_blank">
Using native and syntenically mapped cDNA alignments to improve de novo gene finding</a>.
<em>Bioinformatics</em>. 2008 Mar 1;24(5):637-44.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/18218656" target="_blank">18218656</a>
</p>
<p>
Stanke M, Steinkamp R, Waack S, Morgenstern B.
<a href="https://www.ncbi.nlm.nih.gov/pubmed/15215400" target="_blank">
AUGUSTUS: a web server for gene finding in eukaryotes</a>.
<em>Nucleic Acids Res</em>. 2004 Jul 1;32(Web Server issue):W309-12.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/15215400" target="_blank">15215400</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC441517/" target="_blank">PMC441517</a>
</p>
